src/mnesia_rocksdb.erl

```diff
@@ -338,21 +339,36 @@ prefixes(_) ->
 %% set is OK as ordered_set is a kind of set.
 check_definition(Alias, Tab, Nodes, Props) ->
     Id = {Alias, Nodes},
-    Props1 = lists:map(
-               fun({type, T} = P) ->
-                       if T==set; T==ordered_set; T==bag ->
-                               P;
-                          true ->
-                               mnesia:abort({combine_error,
-                                             Tab,
-                                             [Id, {type,T}]})
-                       end;
-                  ({user_properties, _} = P) ->
-                       %% should perhaps verify rocksdb options...
-                       P;
-                  (P) -> P
-               end, Props),
-    {ok, Props1}.
+    try lists:map(
+          fun({type, T} = P) ->
+                  if T==set; T==ordered_set; T==bag ->
+                          P;
+                     true ->
+                          mnesia:abort({combine_error,
+                                        Tab,
+                                        [Id, {type,T}]})
+                  end;
+             ({user_properties, UPs} = P) ->
+                  RdbOpts = proplists:get_value(rocksdb_opts, UPs, []),
+                  case proplists:get_value(on_write_error, RdbOpts) of
```

Comment:
By using the same default here as in the type specification, you could get rid of a case statement here.